### DISCUSSION TITLE
Using .Root for conditional in path segment

### DISCUSSION THREAD

**Discussion by maru-sama** [JanDeDobbeleer/oh-my-posh#1444] [Troubleshoot]
Hello,

Running in bash I would like to change the background color of the path segment, when I am in a root shell. Initially I thought that this should be quite easy with the background_template and a check against .Root but it did not work. Reading through the documentation and looking at the code I saw that apparently the .Root conditional is NOT available in the path module, could this be added?

I know that I can just use two different theme files, one for my user and one for the root user, but this means whenever I make a change I have to change both files which I tend to forget. :)

**Comment by maru-sama** [comment]
I had another look at this and apparently pull #1404 should make this possible, becaue templating is then supported also for the path segment.

**Reply by JanDeDobbeleer** [reply]
@maru-sama yes, but the implementation needs an update. I'll see if I can quickly add this myself 👍🏻

**Reply by JanDeDobbeleer** [reply]
@maru-sama templates are available now

**Comment by maru-sama** [comment]
@JanDeDobbeleer sorry I was a little bit too fast. The template works just fine but {{.Root}} is not available in the foreground_templates setting so I cannot use it as a conditional

**Reply by JanDeDobbeleer** [reply]
That's indeed a different feature request. I'll adjust the template engine there.

**Reply by maru-sama** [reply]
Ok, I thought Root is available in general in all the templates, but now I see that it is only available in the Session segment for now. 
Does this have a negative impact on the performance, or is .Root always calculated and you only need to make it available in the Path segment?

**Reply by JanDeDobbeleer** [reply]
It's possible to add it, there's for sure some performance overhead which I can calculate.

**Reply by maru-sama** [reply]
Actually adding it is very straight forward now that the template engine is available in the path segment. I wonder if there is a way to cache these values across segments and not per segment.

**Reply by JanDeDobbeleer** [reply]
@maru-sama I'm curious about the use-case as I can do it, but it will clash with existing contexts meaning I'm not inclined to add it. What's your exact use-case?

**Reply by maru-sama** [reply]
My goal was to have one theme that I can use both for my normal user account and for the time I am in a su shell. If I can conditionally check for this I only have one theme instead of two different ones. That said with the new template functionality available I added the .Root value to the path segment in my fork so I am covered. I know that this is a very specific use case, so I am totally fine to keep it in my loacl version only.

**Reply by JanDeDobbeleer** [reply]
But then you want to change colors, right? For all segments or only path? I take it the session/root segment aren't sufficient?

**Reply by maru-sama** [reply]
With my current theme only Session and path are really needed, but of course it would be nice if .Root would be available in all Segments.

**Comment by maru-sama** [comment]
For someone else who is reading this, I ended up extending my personal version of oh-my-posh to include the .Root value also in the path segment.

**Reply by JanDeDobbeleer** [reply]
@maru-sama see #1565 I'm test driving this a bit more before perming it but so far so good (and I was able to find a clean way, it's also faster now)

**Comment by maru-sama** [comment]
Hi, this is very good news, I will give it a spin as well
